src/many-types/src/memo.rs

```diff
@@ -131,6 +139,17 @@ impl<const M: usize> Memo<M> {
     }
 }
 
+// This helps comparisons.
+impl<const M: usize> PartialEq<str> for Memo<M> {
+    fn eq(&self, other: &str) -> bool {
+        if self.len() == 1 {
```

Comment:
Do we really need this check?

Reply:
Yes. If there's more than 1 string, `== "some string"` should always be false.